pkg/volume/glusterfs/glusterfs.go

```diff
@@ -255,8 +260,17 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	for hostIP := range addr {
 		errs = b.mounter.Mount(hostIP+":"+b.path, dir, "glusterfs", options)
 		if errs == nil {
+			glog.Infof("glusterfs: successfully mounted %s", dir)
 			return nil
 		}
 	}
+
+	// Failed mount scenario.
+	// Since gluster does not return eror text
+	// it all goes in a log file, we will read the log file
+	logerror := readGlusterLog(log, b.pod.Name)
+	if logerror != nil {
```

Comment:
if no `/sbin/mount.glusterfs` is on your node, mount will fail (as reported by `errs`) and `readGlusterLog` will return an error (failed to open log file), but since `errs` is not returned, the real cause is not reported .